### PR TITLE
reorg smartcache docs, fixup links, etc.

### PR DESCRIPTION
**Pull request by bpkroth**
Follow ups to https://github.com/microsoft/MLOS/pull/101#discussion_r496232172

### CODE REVIEW

File: source/Mlos.Agent.Server/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+The `Mlos.Agent.Server` has several different operating modes.
+Please see the `--help` usage output for additional detail.s
+
+```shell
+dotnet target/bin/Release/Mlos.Agent.Server.dll --help
+```
+
+## Caveats
+
+- The system currently only supports one shared memory region and doesn't always cleanup the shared memory after itself if something goes wrong.
```
Comment (amueller) [inline source/Mlos.Agent.Server/README.md:63]: A list with a single item probably looks a bit strange but I don't have strong feelings about it.

File: source/Mlos.Agent.Server/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+### Windows
+
+```cmd
+msbuild /m /r source/Mlos.Agent.Server/Mlos.Agent.Server.csproj
+```
+
+## Executing
+
+The `Mlos.Agent.Server` has several different operating modes.
+Please see the `--help` usage output for additional detail.s
```
Comment (amueller) [inline source/Mlos.Agent.Server/README.md:55]:
```suggestion
Please see the `--help` usage output for additional details.
```

File: source/Mlos.Agent.Server/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Shared memory communication channels via the [`Mlos.Agent`](../Mlos.Agent/#mlos-github-tree-view) and [`Mlos.NetCore`](../Mlos.NetCore/#mlos-github-tree-view) libraries.
+
+2. A [`Mlos.Agent.GrpcServer`](../Mlos.Agent.GrpcClient/#mlos-github-tree-view) GRPC channel to allow driving the experimentation process from a Jupyter notebook (work in progress).
+
+3. A GRPC client to allow connecting to the (Python) [`mlos.Grpc.OptimizerMicroserviceServer`](../Mlos.Python/mlos/Grpc/OptimizerMicroserviceServer.py#mlos-github-tree-view) to store and track those experiments.
+
+Since it is meant as a reusable agent for different components, it contains no specific message processing logic itself.
+
+Rather, it starts an [`Mlos.Agent`](../Mlos.Agent/#mlos-github-tree-view) message processing loop which loads each component's SettingsRegistry assembly dlls upon registration request (via the `RegisterComponentConfig` and `RegisterAssemblyRequestMessage` from [`Mlos.Core`](../Mlos.Core/#mlos-github-tree-view) and [`Mlos.NetCore`](../Mlos.NetCore/#mlos-github-tree-view)) and runs their `AssemblyInitializer` constructor to setup the component specific message handler callbacks.
```
Comment (amueller) [inline source/Mlos.Agent.Server/README.md:28]:
```suggestion
Rather, it starts an [`Mlos.Agent`](../Mlos.Agent/#mlos-github-tree-view) message processing loop which loads each component's `SettingsRegistry` assembly dlls upon registration request (via the `RegisterComponentConfig` and `RegisterAssemblyRequestMessage` from [`Mlos.Core`](../Mlos.Core/#mlos-github-tree-view) and [`Mlos.NetCore`](../Mlos.NetCore/#mlos-github-tree-view)) and runs their `AssemblyInitializer` constructor to setup the component specific message handler callbacks.
```

Not sure if "assembly" is a name here and should be code-formatted or capitalized? I wasn't familiar with the term.

File: documentation/RepoOrganization.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ Some notes on the directory layout organization in this repo.
 
     - [Mlos Settings System Code Generation System](../source/Mlos.SettingsSystem.CodeGen/)
     - [Mlos.Core Shared Memory Communication Channel](../source/Mlos.Core/doc/)
+    - [Mlos.Agent.Server](../source/Mlos.Agent.Server/)
```
Comment (amueller) [inline documentation/RepoOrganization.md:19]: Should it also be in the website menu?

File: source/Examples/SmartCache/README.md [hunk 5]
```diff
@@ -6,108 +6,147 @@ It implements a simple cache with different replacement policies and cache size
...
+### Without an optimizer
 
-    ```shell
-    docker run -it -v $PWD:/src/MLOS --name mlos-build mlos/build:ubuntu-20.04
-    ```
+#### Linux
 
-3. Inside the container, [build the compiled software](../../../documentation/02-Build.md#cli-make) with `make`:
+```sh
+dotnet target/bin/Release/Mlos.Agent.Server.dll \
```
Comment (amueller) [inline source/Examples/SmartCache/README.md:57]: this actually fits on a single line, right? that would make copy&paste a bit easier.

File: source/Mlos.Agent.Server/README.md (hunk 1 above)
Comment (bpkroth) [reply source/Mlos.Agent.Server/README.md:63]: Planning for the future ;)

File: documentation/RepoOrganization.md (hunk 4 above)
Comment (bpkroth) [reply documentation/RepoOrganization.md:19]: Should we add a "Components Documentation" section and list this under that?

File: source/Examples/SmartCache/README.md (hunk 5 above)
Comment (bpkroth) [reply source/Examples/SmartCache/README.md:57]: Sure, will fix.

File: source/Mlos.Agent.Server/README.md (hunk 3 above)
Comment (bpkroth) [reply source/Mlos.Agent.Server/README.md:28]: I wasn't either when I first started, that's why I tend to couple it with the term "dll" to make the idea more clear.

File: documentation/RepoOrganization.md (hunk 4 above)
Comment (bpkroth) [reply documentation/RepoOrganization.md:19]: 642b39d1acc9e07b62e37f67ca32f0c3905b0c37 adds this

**Comment by amueller** [comment]
what's the state of this PR?

**Comment by bpkroth** [comment]
> what's the state of this PR?

Was just waiting on another read from you I think.

**Review by amueller — APPROVED** [review]
lgtm apart from the merge issue.

**Pull request merged**